components/datetime/src/neo_marker.rs

```diff
@@ -544,6 +544,12 @@ impl From<NeverField> for Option<CustomTimeZone> {
     }
 }
 
+/// A trait associating [`NeoComponents`].
```

Comment:
nit: docs

Reply:
and mention compile-time somewhere

Reply:
Changed the trait to be named `HasConstComponents`

As far as docs go, I want to do a full overhaul, but I want to get any more feedback from the TC first, which I plan to do tomorrow.